pkg/vendir/fetch/git/git.go

```diff
@@ -149,7 +149,7 @@ func (t *Git) fetch(dstPath string, tempArea ctlfetch.TempArea) error {
 		{"remote", "add", "origin", gitURL},
 	}
 
```

Comment:
I'd say let's have it look like this instead 

```
        ref, err := t.resolveRef(dstPath)
	if err != nil {
		return err
	}
	
        // fetch tags for selection
	argss = append(argss, []string{"config", "remote.origin.tagOpt", "--tags"})

        {
		fetchArgs := []string{"fetch", "origin"}
		if strings.HasPrefix(ref, "origin/") {
			// only fetch the exact ref we're seeking
			fetchArgs = append(fetchArgs, ref[7:])
		}
		if t.opts.Depth > 0 {
			fetchArgs = append(fetchArgs, "--depth", strconv.Itoa(t.opts.Depth))
		}
		argss = append(argss, fetchArgs)
	}

        .... REMOVE `t.resolveRef(dstPath)` from lines 174-177
```

Without any testing it out, I think this will provide us 

a) the opts != nil guards we want.
b) does nothing in the case you select `RefSelection` -  I don't believe `RefSelection` even resolves to a non-semver version (i.e it won't have origin/ in it)

Reply:
Nevermind, we can't list tags first before we fetch them. Scratch this idea.

Reply:
Yeah, I was wondering if there were a way that we could proactively check a single tag, namely whatever's in `Ref`. Maybe something like `git tag -l $(ref)`?

Reply:
But again I guess we'd need to fetch them all first :/